src/OpenTelemetry.Instrumentation.EventCounters/EventCounterListener.cs

```diff
@@ -0,0 +1,229 @@
+
+        protected override void OnEventSourceCreated(EventSource source)
+        {
+            // TODO: Add Configuration options to selectively subscribe to EventCounters
+            try
+            {
+                if (!this.isInitialized)
+                {
+                    this.eventSources.Add(source);
+                }
+                else
+                {
+                    this.EnableEvents(source, EventLevel.Verbose, EventKeywords.All, this.EnableEventArgs);
+                }
+            }
+            catch (Exception ex)
+            {
+                EventCountersInstrumentationEventSource.Log.ErrorEventCounter(source.Name, ex.Message);
+            }
+        }
+
+        protected override void OnEventWritten(EventWrittenEventArgs eventData)
+        {
+            if (!this.isInitialized || !eventData.EventName.Equals("EventCounters"))
+            {
+                return;
+            }
+
+            try
+            {
+                if (eventData.Payload.Count > 0 && eventData.Payload[0] is IDictionary<string, object> eventPayload)
+                {
+                    this.ExtractAndPostMetric(eventData.EventSource.Name, eventPayload);
+                }
+                else
+                {
+                    EventCountersInstrumentationEventSource.Log.IgnoreEventWrittenAsEventPayloadNotParseable(eventData.EventSource.Name);
+                }
+            }
+            catch (Exception ex)
+            {
+                EventCountersInstrumentationEventSource.Log.ErrorEventCounter(eventData.EventName, ex.ToString());
+            }
+        }
+
+        private void ExtractAndPostMetric(string eventSourceName, IDictionary<string, object> eventPayload)
+        {
+            try
+            {
+                bool calculateRate = false;
+                string actualValue = string.Empty;
+
+                string counterName = string.Empty;
+                string counterDisplayName = string.Empty;
+                InstrumentType instrumentType = InstrumentType.Counter;
+
+                foreach (KeyValuePair<string, object> payload in eventPayload)
+                {
+                    var key = payload.Key;
+                    if (key.Equals("Name", StringComparison.OrdinalIgnoreCase))
+                    {
+                        counterName = payload.Value.ToString();
+                    }
+                    else
+                    if (key.Equals("DisplayName", StringComparison.OrdinalIgnoreCase))
+                    {
+                        counterDisplayName = payload.Value.ToString();
+                    }
+                    else if (key.Equals("Mean", StringComparison.OrdinalIgnoreCase))
+                    {
+                        instrumentType = InstrumentType.Counter;
+                        actualValue = payload.Value.ToString();
+                    }
+                    else if (key.Equals("Increment", StringComparison.OrdinalIgnoreCase))
```

Comment:
This would be the incrementingeventcounter, which would be more like OTel's Counter. so this would map to ObservableCounter.
(need to write a unit test to fully see how this whole things works end2end. The EventCounter terminology can be very confusing!)

Reply:
It would appear so and I followed your suggestion but I am not sure this is the right instrument. I feel like this should be an  observable guage as well, because the eventcounterListener always sees a sum aggregation. But if we use the ObservableCounter and set the metric reader options to Delta, these values dont make sense. As it is if we use cumulative, it does make sense.